core/src/main/java/org/infinispan/functional/impl/ReadOnlyMapImpl.java

```diff
@@ -67,7 +69,7 @@ private ReadOnlyMapImpl(Params params, FunctionalMapImpl<K, V> functionalMap) {
    @Override
    public Traversable<ReadEntryView<K, V>> entries() {
       log.tracef("Invoked entries(%s)", params);
-      CloseableIterator<CacheEntry<K, V>> it = fmap.cache.cacheEntrySet().iterator();
+      Iterator<CacheEntry<K, V>> it = fmap.cache.cacheEntrySet().iterator();
```

Comment:
Oops, this is a risky spot; the Traversable is not guaranteed to go through all entries and noone's going to call it.close(). Could you file a JIRA and add TODO referencing the JIRA about a possible leak of `Closeable`? We might need to make `Traversable` `Closeable` as well...
@wburns Any chance I am panicking here?

Reply:
No, @rvansa you are correct. Good catch.

In this case resources would be left open if the Traversable was not used or if a short circuited terminal operation terminated early (findAny, noneMatch, anyMatch, allMatch).

Reply:
Also it seems that this creation of the Iterator is unneeded as we should be able to pass the Stream `fmap.cache.cacheEntrySet().stream()` directly in. Changing the spliterator Characteristics shouldn't matter as the underlying one is already CONCURRENT and we enforce no write updates in the map method below.

Reply:
@wburns Would a direct use of stream change anything about closeability?

Reply:
No, we would just have to close the Stream instead. The second change is about just less wrapping going on :)

Reply:
@wburns @rvansa do you prefer a new ticket for this, or the ticket for tx cache support in functional map can be enough ? I don't see the size of the impact,  don't know if I create a new ticket for this, or not ...

Reply:
@rvansa created https://issues.jboss.org/browse/ISPN-8378